src/validator.ts

```diff
@@ -13,33 +14,62 @@ export function validate(contractNode: NodeToProcess, natspec: Natspec): string[
         return alerts;
     };
 
-    let functionParameters = contractNode.parameters?.parameters.map(p => p.name) ?? [];
-    let natspecParameters = natspec.params.map(p => p.name);
+    if(node instanceof EnumDefinition) {
+        // TODO: Process enums
+    } else if(node instanceof ErrorDefinition) {
+        alerts = [...alerts, ...validateParameters(node, natspec)];
+    } else if(node instanceof EventDefinition) {
+        alerts = [...alerts, ...validateParameters(node, natspec)];
+    } else if(node instanceof FunctionDefinition) {
+        alerts = [...alerts, ...validateParameters(node, natspec)];
+
+        let functionReturns = node.vReturnParameters.vParameters.map(p => p.name);
+        let natspecReturns = natspec.returns.map(p => p.name);
+
+        // Make sure all defined returns have natspec
+        for(let paramName of functionReturns) {
+            if(!natspecReturns.includes(paramName)) {
+                let message = paramName === '' ? '@return missing for unnamed return' : `@return ${paramName} is missing`;
+                alerts.push(message);
+            }
+        }
     
-    // Make sure all defined function parameters have natspec
-    for(let paramName of functionParameters) {
-        if(!natspecParameters.includes(paramName)) {
-            alerts.push(`@param ${paramName} is missing`);
+        // Make sure there is no natspec defined for non-existing returns
+        for(let paramName of natspecReturns) {
+            if(paramName && !functionReturns.includes(paramName)) {
+                alerts.push(`Missing named return for: @return ${paramName}`);
+            }
+        }
```

Comment:
can we wrap this in functions to keep the code cleaner? one fn could be `validateReturnParameters`

Reply:
same with the StructDefinition